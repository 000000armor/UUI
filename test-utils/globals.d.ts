declare const __DEV__: boolean;
declare let __PACKAGE_VERSION__: string;

declare module 'jsurl';
declare module 'query-string';
