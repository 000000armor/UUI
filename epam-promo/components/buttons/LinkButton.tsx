import { devLogger, withMods } from '@epam/uui-core';
import { LinkButton as UuiLinkButton, LinkButtonProps as UuiLinkButtonProps } from '@epam/uui';

export interface LinkButtonMods {
    color?: 'blue' | 'green' | 'amber' | 'red' | 'gray60' | 'gray10';
}

export type LinkButtonProps = Omit<UuiLinkButtonProps, 'color'> & LinkButtonMods;

export const LinkButton = withMods<Omit<UuiLinkButtonProps, 'color'>, LinkButtonMods>(
    UuiLinkButton,
    () => [],
    (props) => {
        devLogger.warnAboutDeprecatedPropValue<LinkButtonProps, 'color'>({
            component: 'LinkButton',
            propName: 'color',
            propValue: props.color,
            condition: () => ['green', 'amber', 'red'].indexOf(props.color) !== -1,
        });
        return {
            color: props.color ?? 'blue',
        } as LinkButtonProps;
    },
);
