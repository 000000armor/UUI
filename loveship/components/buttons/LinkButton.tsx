import * as types from '../types';
import { withMods } from '@epam/uui-core';
import { LinkButton as UuiLinkButton, LinkButtonProps } from '@epam/uui';
import { TextSettings } from "../../helpers/textLayout";

export interface LinkButtonMods extends TextSettings {
    color?: types.EpamColor;
    font?: types.FontStyle;
}

export const LinkButton = withMods<LinkButtonProps, LinkButtonMods>(UuiLinkButton, () => {});
