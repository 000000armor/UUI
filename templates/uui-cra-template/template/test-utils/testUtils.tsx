import * as uuiTestUtils from '@epam/uui-test-utils';
import { ReactNode, ReactElement, useState, useEffect } from "react";
import { TAppContext } from "../src/data/appContext";
import { StubAdaptedRouter, useUuiServices, UuiContext } from "@epam/uui-core";
import { TApi } from "../src/data/apiDefinition";
import { svc } from "../src/services";

const testAppContext: TAppContext = {};
const testApi: TApi = {
    loadLinksForMainPage: async () => [
        {
            label: 'UUI docs: ',
            link: 'https://uui.epam.com/',
            linkLabel: 'uui.epam.com',
        },
        {
            label: 'Git: ',
            link: 'https://github.com/epam/uui',
            linkLabel: 'github.com/epam/uui',
        },
    ],
};

const router = new StubAdaptedRouter();
function TestContextWrapper({ children }: { children?: ReactNode }) {
    const [isLoaded, setIsLoaded] = useState(false);
    const { services } = useUuiServices<TApi, TAppContext>({ apiDefinition: () => testApi, router });

    useEffect(() => {
        Promise.resolve(testAppContext).then((appCtx) => {
            services.uuiApp = appCtx;
            Object.assign(svc, services);
            setIsLoaded(true);
        })
    }, [services]);

    if (isLoaded) {
        return (
            <UuiContext.Provider value={ services }>
                {children}
            </UuiContext.Provider>
        );
    }
    return null;
}

/**
 *  re-export UUI test utils 'as is' and override some methods with custom test context
 */
export * from '@epam/uui-test-utils';

// Override method from UUI test utils
export const renderWithContextAsync = async (reactElement: ReactElement) => {
    return uuiTestUtils.renderWithContextAsync(reactElement, {
        wrapper: TestContextWrapper as any,
    });
};
