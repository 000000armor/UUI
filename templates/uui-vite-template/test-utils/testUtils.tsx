import * as uuiTestUtils from '@epam/uui-test-utils';
import { ReactNode, ReactElement } from 'react';
import { TApi } from '../src/app/helpers/apiDefinition';
import { TAppContext } from '../src/app/helpers/appContext';
import { ContextProvider } from '@epam/uui-core';

const testAppContext: TAppContext = {};
const testApi: TApi = {
    loadLinksForMainPage: async () => [
        {
            label: 'UUI docs: ',
            link: 'https://uui.epam.com/',
            linkLabel: 'uui.epam.com',
        },
        {
            label: 'Git: ',
            link: 'https://github.com/epam/uui',
            linkLabel: 'github.com/epam/uui',
        },
    ],
};
function TestContextWrapper({ children }: { children?: ReactNode }) {
    return (
        <ContextProvider<TApi, TAppContext>
            onInitCompleted={() => {}}
            apiDefinition={() => testApi}
            loadAppContext={() => Promise.resolve(testAppContext)}
        >
            {children}
        </ContextProvider>
    );
}

/**
 *  re-export UUI test utils 'as is' and override some methods with custom test context
 */
export * from '@epam/uui-test-utils';

// Override method from UUI test utils
export const renderWithContextAsync = async (reactElement: ReactElement) => {
    return uuiTestUtils.renderWithContextAsync(reactElement, {
        wrapper: TestContextWrapper,
    });
};
