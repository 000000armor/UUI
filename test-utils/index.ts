export * from './src/extensions/testingLibraryReactExt';
export * from './src/mockUtils/adaptivePanelMockUtils';
export * from './src/mockUtils/reactPortalsMockUtils';
export * from './src/mockUtils/svgrMock';
export * from './src/utils/renderingWithContextUtils';
export * from './src/utils/setupComponentUtils';
export * from './src/utils/timerUtils';
