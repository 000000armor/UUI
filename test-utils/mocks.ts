export const windowMock = {
    matchMedia: () => ({matches: false}),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
} as any;