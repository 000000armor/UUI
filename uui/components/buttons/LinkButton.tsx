import { withMods } from '@epam/uui-core';
import { Button, ButtonProps } from '@epam/uui-components';
import { getIcon, icons } from '../../icons';
import { getIconClass } from './helper';
import * as css from './LinkButton.scss';
import '../../assets/styles/variables/buttons/linkButton.scss';
import { uuiComponentClass } from '../constant';

const defaultSize = '36';

export interface LinkButtonMods {
    size?: string;
}

function applyLinkButtonMods(mods: LinkButtonMods & ButtonProps) {
    return [
        uuiComponentClass.linkButton,
        'link-button-vars',
        css.root,
        css['size-' + (mods.size || defaultSize)],
        ...getIconClass(mods),
    ];
}

export const LinkButton = withMods<ButtonProps, LinkButtonMods>(Button, applyLinkButtonMods, (props) => ({
    dropdownIcon: getIcon('foldingArrow'),
    clearIcon: getIcon('clear'),
}));