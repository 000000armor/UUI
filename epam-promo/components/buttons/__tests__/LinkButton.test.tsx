import React from 'react';
import { LinkButton } from '../LinkButton';
import { SvgMock, renderSnapshotWithContextAsync } from '@epam/test-utils';

describe('LinkButton', () => {
    it('should render with minimum props', async () => {
        const tree = await renderSnapshotWithContextAsync(<LinkButton />);
        expect(tree).toMatchSnapshot();
    });

    it('should render with maximum props', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <LinkButton color="blue" onClick={ jest.fn } icon={ SvgMock } isDisabled={ false } isDropdown onClear={ jest.fn } />,
        );
        expect(tree).toMatchSnapshot();
    });
});
