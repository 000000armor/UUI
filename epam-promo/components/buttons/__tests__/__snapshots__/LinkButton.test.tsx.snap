// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton should render with maximum props 1`] = `
<button
  aria-disabled={false}
  className="container uui-button-box uui-enabled -clickable root size-36 uui-has-left-icon uui-has-right-icon"
  onClick={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
  <div
    className="container uui-icon uui-enabled -clickable -clickable"
    onClick={[Function]}
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
</button>
`;

exports[`LinkButton should render with minimum props 1`] = `
<button
  className="container uui-button-box uui-enabled root size-36 uui-no-left-icon uui-no-right-icon"
  onClick={[Function]}
  tabIndex={-1}
  type="button"
/>
`;
